assets/src/js/query_loop_carousel.js

```diff
@@ -21,70 +29,76 @@ export const setupQueryLoopCarousel = () => {
       list.after(carousel);
       carousel.append(list);
 
-      const indicators = document.createElement('ol');
-      indicators.classList.add('carousel-indicators');
-      carousel.append(indicators);
+      if (!list) {
+        return;
+      }
+      const posts = list.querySelectorAll('.wp-block-post');
+
+      // Only add indicators if there are more items to show
+      if (posts.length > itemsPerSlide) {
+        indicators = document.createElement('ol');
+        indicators.classList.add('carousel-indicators');
+        carousel.append(indicators);
+
+        // Update controls
+        ARROW_DIRECTIONS.forEach(direction => {
+          const controlBtn = layout.querySelector(`.wp-block-buttons .carousel-control-${direction}`);
+          controlBtn.dataset.bsTarget = `#${uniqueId}`;
+          controlBtn.dataset.bsSlide = direction;
```

Comment:
If `controlBtn` is not found, accessing `controlBtn.dataset` or `controlBtn.querySelector` will throw an error.